.travis.yml

```diff
@@ -135,22 +129,19 @@ before_install:
        /usr/bin/c++ --version
      fi
    - |
-     if [ "$xRUN_BUILD_AND_TESTSUITE" = "1" ]; then
+     if [ "$xNEST_BUILD_TYPE" != "STATIC_CODE_ANALYSIS" ]; then
          echo "+ + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + +"
          echo "+         P R E P A R E   N E S T   B U I L D   E N V I R O N M E N T         +"
          echo "+ + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + +"
-         echo ${MATRIX_EVAL}
-         eval "${MATRIX_EVAL}"
          cp extras/install_csa-libneurosim.sh $HOME
          cp extras/install_music.sh $HOME
          cp extras/install_sionlib.sh $HOME
          cp extras/install_libboost.sh $HOME
          cd $HOME/build
          echo $PATH
-
-         if [ MATRIX_EVAL="CC=clang-7 && CXX=clang++-7" ]; then
-           pyenv install --list
-           pyenv global 3.6.10
+         if [ "$xNEST_BUILD_COMPILER" = "CLANG" ]; then
+             pyenv install --list
+             pyenv global 3.6.10
```

Comment:
Hard coded version number: potentially an issue?

Reply:
Agreed. I tried removing the lines altogether in ec26ee940. Let's see what Travis thinks of this.

Reply:
OK. Apparently it [picks up Python 2.7](https://travis-ci.org/github/nest/nest-simulator/jobs/726840827#L1289) without these lines. I have therefore removed the commit again in my last push.

One alternative could be to define the exact version number as an environment variable in [line 60](https://github.com/nest/nest-simulator/pull/1704/files/a6026390f0dd42d7d14e69545201d2087a890085#diff-354f30a63fb0907d4ad57269548329e3R60), but I'm afraid we would then still have the setting twice, once for our purposes in line 60 and once to tell Travis what we want in [line 59](https://github.com/nest/nest-simulator/pull/1704/files/a6026390f0dd42d7d14e69545201d2087a890085#diff-354f30a63fb0907d4ad57269548329e3R59). The advantage would just be code locality. What do you think?

Reply:
Could this trick be used?

```bash
$ python3 -c "import sys; print('%d.%d.%d' % (sys.version_info.major, sys.version_info.minor, sys.version_info.micro))"
3.7.3
```

Reply:
Maybe. In 32365933d, I'm trying out a version based on the somewhat shorter `python3 --version | cut -d' ' -f2`.

Reply:
OK. After that also [failed](https://travis-ci.org/github/nest/nest-simulator/jobs/727040026#L947), I have now implemented the solution with the additional environment variable described above. This seems to be working fine.